components/ai_chat_ui/components/conversation_list/index.tsx

```diff
@@ -8,40 +8,27 @@ import classnames from 'classnames'
 
 import styles from './style.module.scss'
 import { ConversationTurn, CharacterType } from '../../api/page_handler'
-import { getLocale } from '$web-common/locale'
 
 interface ConversationListProps {
   list: ConversationTurn[]
   isLoading: boolean
 }
 
-const elementScrollBehavior: ScrollIntoViewOptions = {
-  behavior: 'smooth',
-  block: 'start'
-}
-
 function ConversationList (props: ConversationListProps) {
   // Scroll the last conversation item in to view when entries are added.
   const lastConversationEntryElementRef = React.useRef<HTMLDivElement>(null)
-  const loadingElementRef = React.useRef<HTMLDivElement>(null)
+
   React.useEffect(() => {
     if (!props.list.length && !props.isLoading) {
       return
     }
-    if (props.isLoading) {
-      if (!loadingElementRef.current) {
-        console.error('Conversation loading element did not exist when expected')
-      } else {
-        loadingElementRef.current.scrollIntoView(elementScrollBehavior)
-      }
-      return
-    }
+
     if (!lastConversationEntryElementRef.current) {
       console.error('Conversation entry element did not exist when expected')
     } else {
-      lastConversationEntryElementRef.current.scrollIntoView(elementScrollBehavior)
+      lastConversationEntryElementRef.current.scrollIntoView(false)
     }
-  }, [props.list.length, props.isLoading])
+  }, [props.list.length, props.isLoading, lastConversationEntryElementRef.current?.clientHeight])
```

Comment:
What's the reasoning on depending on a ref's child attribute? Changing the value of that won't trigger a react render.

Reply:
The effect logic here aims to interact with the browser's scroll event rather than triggering a re-render. React will re-render to update the DOM, which can result in changes to the height of the `<div>`. When such changes occur, the effect should respond by triggering the browser's scroll event.

Perhaps, a this can be better using a resize observer.